### ISSUE TITLE
SC2039, printf: -F at the beginning of a format string is recognized as an option

### ISSUE THREAD

**Issue by hcpl** [koalaman/shellcheck#776]
The question is: how could this even be interpreted as an option?

As for a background: the issue is about constructing a command based on information that is only known at run time. My use case involves generating a list of `name=value` pairs that preserve spaces in a POSIX compliant way.
Specifically for `curl`:
```sh
curl -fsSL -F "$formname1=$content1" -F "$formname2=$content2" ... "$URL"
```
UPD. Related to #380. As someone commented there, using quoted strings to represent possibly space containing values is a bad practice. POSIX `sh` doesn't have arrays, so we have to use functions and positional parameters. But is this really the only viable solution? (sorry if the actual issue became offtopic for this project).

#### For bugs
- Rule Id: SC2039
- `shellcheck --version`:
```
ShellCheck - shell script analysis tool
version: 0.4.5
license: GNU General Public License, version 3
website: http://www.shellcheck.net
```
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:
```sh
#!/bin/sh
printf -- '-F %s' foo
```

#### Here's what shellcheck currently says:
```
In test.sh line 2:
printf -- '-F %s' foo
          ^-- SC2039: In POSIX sh, printf -F is undefined.
```
#### Here's what I wanted or expected to see:
```sh
```

**Comment by koalaman** [comment]
You're right, this is most definitely a bug. No warning should be given for flags after `--`. 

From my reading of POSIX, `--` shouldn't even be accepted, but I didn't find a shell where this was problematic.

Functions and positional parameters are a good solution because it's the one array `sh` supports, but it's also possible to e.g. pick a delimiter you won't support in values and use word splitting (while disabling globs), or to very carefully escape the string in a shell independent way and eval it.

**Comment by koalaman** [comment]
Fixed in f6618d4
